[2023-08-08T11:05:30.724+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ETL_Pipeline.transform_data scheduled__2023-08-08T10:00:00+00:00 [queued]>
[2023-08-08T11:05:30.733+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ETL_Pipeline.transform_data scheduled__2023-08-08T10:00:00+00:00 [queued]>
[2023-08-08T11:05:30.733+0000] {taskinstance.py:1308} INFO - Starting attempt 2 of 2
[2023-08-08T11:05:30.748+0000] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): transform_data> on 2023-08-08 10:00:00+00:00
[2023-08-08T11:05:30.754+0000] {standard_task_runner.py:57} INFO - Started process 324 to run task
[2023-08-08T11:05:30.759+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'ETL_Pipeline', 'transform_data', 'scheduled__2023-08-08T10:00:00+00:00', '--job-id', '5047', '--raw', '--subdir', 'DAGS_FOLDER/my_first_dag.py', '--cfg-path', '/tmp/tmpcifa64n_']
[2023-08-08T11:05:30.762+0000] {standard_task_runner.py:85} INFO - Job 5047: Subtask transform_data
[2023-08-08T11:05:30.807+0000] {task_command.py:410} INFO - Running <TaskInstance: ETL_Pipeline.transform_data scheduled__2023-08-08T10:00:00+00:00 [running]> on host 48849dae508a
[2023-08-08T11:05:30.901+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='ETL_Pipeline' AIRFLOW_CTX_TASK_ID='transform_data' AIRFLOW_CTX_EXECUTION_DATE='2023-08-08T10:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-08-08T10:00:00+00:00'
[2023-08-08T11:05:30.905+0000] {logging_mixin.py:150} WARNING - Scraping Progress:   0%|          | 0/10 [00:00<?, ?it/s]
[2023-08-08T11:05:35.065+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  10%|#         | 1/10 [00:04<00:37,  4.16s/it]
[2023-08-08T11:05:36.971+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  20%|##        | 2/10 [00:06<00:22,  2.83s/it]
[2023-08-08T11:05:38.102+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  30%|###       | 3/10 [00:07<00:14,  2.06s/it]
[2023-08-08T11:05:40.588+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  40%|####      | 4/10 [00:09<00:13,  2.23s/it]
[2023-08-08T11:05:41.536+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  50%|#####     | 5/10 [00:10<00:08,  1.76s/it]
[2023-08-08T11:05:44.778+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  60%|######    | 6/10 [00:13<00:09,  2.27s/it]
[2023-08-08T11:05:46.835+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  70%|#######   | 7/10 [00:15<00:06,  2.20s/it]
[2023-08-08T11:05:47.501+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  80%|########  | 8/10 [00:16<00:03,  1.71s/it]
[2023-08-08T11:05:48.108+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  90%|######### | 9/10 [00:17<00:01,  1.37s/it]
[2023-08-08T11:05:57.054+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  90%|######### | 9/10 [00:26<00:02,  2.91s/it]
[2023-08-08T11:05:57.054+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/my_first_dag.py", line 40, in transform_data
    data_list.append(scrape_data_from_url(url))
  File "/opt/airflow/dags/utils/scraper.py", line 8, in scrape_data_from_url
    article.parse()
  File "/home/airflow/.local/lib/python3.7/site-packages/newspaper/article.py", line 191, in parse
    self.throw_if_not_downloaded_verbose()
  File "/home/airflow/.local/lib/python3.7/site-packages/newspaper/article.py", line 532, in throw_if_not_downloaded_verbose
    (self.download_exception_msg, self.url))
newspaper.article.ArticleException: Article `download()` failed with HTTPSConnectionPool(host='www.prothomalo.com', port=443): Read timed out. on URL https://www.prothomalo.com/video/bangladesh/klvtd7jbfa
[2023-08-08T11:05:57.067+0000] {taskinstance.py:1350} INFO - Marking task as FAILED. dag_id=ETL_Pipeline, task_id=transform_data, execution_date=20230808T100000, start_date=20230808T110530, end_date=20230808T110557
[2023-08-08T11:05:57.086+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 5047 for task transform_data (Article `download()` failed with HTTPSConnectionPool(host='www.prothomalo.com', port=443): Read timed out. on URL https://www.prothomalo.com/video/bangladesh/klvtd7jbfa; 324)
[2023-08-08T11:05:57.115+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-08-08T11:05:57.138+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
